cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DefaultTaintTrackingImpl.qll

```diff
@@ -54,15 +55,31 @@ predicate predictableOnlyFlow(string name) {
     ]
 }
 
+class DefaultTaintTrackingOptions extends string {
```

Comment:
Needs QLDoc even though its internal.

My interpretation is that this is a new class to replace uses of `SecurityOptions.isUserInput` (planned for deprecation) with a similar interface but implemented using modern taint sources.  Its only intended for internal use and only temporarily - external queries should be modernized away from `DefaultTaintTracking` entirely during the deprecation cycle?

Reply:
Correct. I started out not having this at all and just having a predicate. Unfortunately we have two queries that extend the original class.